engine/pkg/externalresource/broker/broker.go

```diff
@@ -138,21 +175,46 @@ func (b *DefaultBroker) createResource(
 
 // OnWorkerClosed implements Broker.OnWorkerClosed
 func (b *DefaultBroker) OnWorkerClosed(ctx context.Context, workerID resModel.WorkerID, jobID resModel.JobID) {
-	scope := internal.ResourceScope{
-		Executor: b.executorID,
-		WorkerID: workerID,
+	timer := time.NewTimer(defaultTimeout)
+	select {
+	case b.closedWorkerCh <- closedWorker{workerID: workerID, jobID: jobID}:
+		return
+	case <-timer.C:
+		log.Error("closed worker channel is full, broker may be stuck")
 	}
-	for _, fm := range b.fileManagers {
-		err := fm.RemoveTemporaryFiles(ctx, scope)
-		if err != nil {
-			// TODO when we have a cloud-based error collection service, we need
-			// to report this.
-			// However, since an error here is unlikely to indicate a correctness
-			// problem, we do not take further actions.
-			log.Warn("Failed to remove temporary files for worker",
-				zap.String("worker-id", workerID),
-				zap.String("job-id", jobID),
-				zap.Error(err))
+}
+
+func (b *DefaultBroker) runGCClosedWorker(ctx context.Context) {
+	// We run a gc loop at the max frequency of once per second.
+	rl := ratelimit.New(1 /* once per second */)
+	for {
+		rl.Take()
+		select {
+		case <-ctx.Done():
+			return
+		case w := <-b.closedWorkerCh:
+			scope := internal.ResourceScope{
+				Executor: b.executorID,
+				WorkerID: w.workerID,
+			}
+			for _, fm := range b.fileManagers {
+				err := fm.RemoveTemporaryFiles(ctx, scope)
+				if err != nil {
```

Comment:
If process crashes, are any temporary files leak?

Reply:
There are no resource leaks, since all temporary resources created by the crashed executor would be cleaned by [gcRunner](https://github.com/pingcap/tiflow/pull/7204/files#diff-3707465866a91b7dcb1b1b26212a10f25e1b13f91f0fefa5b6ffd266e4c5851bR228).